### PR TITLE
Redux slices

### PR DESCRIPTION
**Pull request by muselesscreator**
Update redux usage to use redux toolkit's createSlice utility, and group redux data accordingly.

As discussed in code-walkthrough.

JIRA: [AU-392](https://openedx.atlassian.net/browse/AU-392)

### CODE REVIEW

File: src/containers/ResponseDisplay/SubmissionFiles.jsx [hunk 1]
```diff
@@ -51,6 +51,7 @@ export class SubmissionFiles extends React.Component {
                   },
                 ]}
                 data={files}
+                itemCount={files.length}
```
Comment (leangseu-edx) [inline src/containers/ResponseDisplay/SubmissionFiles.jsx]: I think I already did this on the other pr.

File: src/data/redux/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  requests,
+  submissions,
+};
+
+const moduleProps = (propName) => Object.keys(modules).reduce(
+  (obj, moduleKey) => ({ ...obj, [moduleKey]: modules[moduleKey][propName] }),
+  {},
+);
+
+/* istanbul ignore next */
```
Comment (leangseu-edx) [inline src/data/redux/index.js]: What is this?

File: src/data/redux/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+export { default as thunkActions } from './thunkActions';
+
+const modules = {
+  app,
+  grading,
+  requests,
+  submissions,
+};
+
+const moduleProps = (propName) => Object.keys(modules).reduce(
```
Comment (leangseu-edx) [inline src/data/redux/index.js:19]: Fancy 👍

**Review by leangseu-edx — APPROVED** [review]
LGTM 👍

File: src/containers/ResponseDisplay/SubmissionFiles.jsx (hunk 1 above)
Comment (muselesscreator) [reply src/containers/ResponseDisplay/SubmissionFiles.jsx]: yep, they will catch in rebase.  Was making sure I cleared all my errors.

File: src/data/redux/index.js (hunk 2 above)
Comment (muselesscreator) [reply src/data/redux/index.js]: tells coverage not to check "put things in an object" code.

**Pull request merged**